bool_slice.go

```diff
@@ -0,0 +1,145 @@
+package pflag
+
+import (
+	"encoding/csv"
+	"io"
+	"strconv"
+	"strings"
+)
+
+// -- boolSlice Value
+type boolSliceValue struct {
+	value   *[]bool
+	changed bool
+}
+
+func newBoolSliceValue(val []bool, p *[]bool) *boolSliceValue {
+	bsv := new(boolSliceValue)
+	bsv.value = p
+	*bsv.value = val
+	return bsv
+}
+
+// Set converts, and assigns, the comma-separated boolean argument string representation as the []bool value of this flag.
+// If Set is called on a flag that already has a []bool assigned, the newly converted values will be appended.
+func (s *boolSliceValue) Set(val string) error {
+
+	// remove all quote characters
+	rmQuote := strings.NewReplacer(`"`, "", `'`, "", "`", "")
+	r := csv.NewReader(strings.NewReader(rmQuote.Replace(val)))
+
+	// read flag arguments with CSV parser
+	boolStrSlice, err := r.Read()
+	if err != nil && err != io.EOF {
```

Comment:
I should we re-use https://github.com/spf13/pflag/blob/master/string_slice.go#L22 and not include CSV at all? Should we use `writeAsCSV()` instead of `strings.Join()`? It will make almost no difference, as you noticed, but I think it would make the code a bit more consistent.